File: frontend/src/lib/disputeStatus.js
const TERMINAL_DISPUTE_STATUSES = new Set([
  "resolved",
  "resolved_contractor",
  "resolved_customer",
  "resolved_homeowner",
  "closed",
  "canceled",
  "cancelled",
]);

export function normalizeDisputeStatus(status) {
  return String(status || "").trim().toLowerCase();
}

export function isDisputeTerminal(status) {
  const normalized = normalizeDisputeStatus(status);
  if (!normalized) return false;
  if (TERMINAL_DISPUTE_STATUSES.has(normalized)) return true;
  return normalized.startsWith("resolved_");
}

export function canRespondToDispute(status) {
  const normalized = normalizeDisputeStatus(status);
  return !isDisputeTerminal(normalized) && (normalized === "open" || normalized === "under_review");
}

export function canCancelDispute(status) {
  const normalized = normalizeDisputeStatus(status);
  return !isDisputeTerminal(normalized) && (normalized === "initiated" || normalized === "open");
}

export function canUploadToDispute(status) {
  return !isDisputeTerminal(status);
}

export function canResolveDispute(status) {
  return !isDisputeTerminal(status);
}

export function isDisputeArchived(dispute) {
  return Boolean(dispute?.is_archived);
}

export function getDisputeReadOnlyLabel(status) {
  return isDisputeTerminal(status) ? "Read only" : "";
}

File: backend/projects/services/dispute_status.py
from __future__ import annotations


TERMINAL_DISPUTE_STATUSES = {
    "resolved",
    "resolved_contractor",
    "resolved_customer",
    "resolved_homeowner",
    "closed",
    "canceled",
    "cancelled",
}


def normalize_dispute_status(value) -> str:
    return str(value or "").strip().lower()


def is_terminal_dispute_status(value) -> bool:
    status = normalize_dispute_status(value)
    if not status:
        return False
    if status in TERMINAL_DISPUTE_STATUSES:
        return True
    return status.startswith("resolved_")


def is_active_dispute_status(value) -> bool:
    return not is_terminal_dispute_status(value)
